wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -393,6 +406,87 @@ function getCardinalNumber(number) {
   }
 }
 
+export function StarOurRepoModal({ isStarRepoOpen, setIsStarRepoOpen, appGenerationResult }) {
+  const [tokens, setTokens] = useState(0)
+  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isStarRepoOpen })
+
+  const tokenNumberStr = appGenerationResult?.project?.logs?.filter(log => log.content.toLowerCase().includes("tokens usage") === true)[0]?.content.split(':')[1]?.trim()
+
+  useEffect(() => {
+    if (!!tokenNumberStr) {
+      const num = tokenNumberStr.slice(1, -1) * 1000
```

Comment:
While the whole parsing of "tokens usage" is fragile due to output possibly changing, this part feels especially fragile, where you count on specific characters being in the right place. Btw what shape are you assuming here? " 12k" or " 3k" or something like that, and you are trying to remove the initial space and the "k" at the end?
In that case, it would be better to make that assumption more clear, e.g. you could use regex that says it expects some white space, then some digits, and then a letter "k", and then takes the number out (matches on it).

Reply:
Btw, what you could do, is detecting if parsing failed (if that thing is not a number), and then just display initial thing instead of parsing it.